src/AbstractBlock.php

```diff
@@ -43,6 +43,8 @@ public function renderBlockCallback(array $block): void
 
         $controller = $this;
 
+        extract($this->with());
```

Comment:
extract() is a terrible function that makes code harder to debug and harder to understand. We should discourage it’s [sic] use and remove all of our uses of it.

https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#dont-extract

Reply:
I see your point, but there is no alternative that I can see to provide the feature.
Do you have any suggestions to work around it?

However (given the reasons to discourage it), surely the fault/issue would fall on the user of the function rather than the function being an issue?